### PR TITLE
Use `Vec::with_capacity()` when we know vec capacity

### PR DESCRIPTION
**Pull request by danlaine**
I ignored usages of `Vec::new()` in tests. You may think that using `std::mem::size_of::<u64>()` (etc.) is overkill. If so, fair enough. If you want, let me know and I'll change it to a good ol' `8` or `4`.

In some instances, you may feel that the extra code/clutter doesn't merit the performance gains. Again, fair enough. In those cases where we care about code aesthetics more than performance, I'd suggest doing, for example:

```rust
     [
        round.to_be_bytes().as_slice(),
        dealer.to_be_bytes().as_slice(),
        share,
    ]
    .concat()
```

rather than

```rust
    let mut payload = Vec::new();
    payload.extend_from_slice(&round.to_be_bytes());
    payload.extend_from_slice(&dealer.to_be_bytes());
    payload.extend_from_slice(share);
    payload
```

I believe these are semantically equivalent (don't quote me on that) and fwiw I think the former is prettier and maybe more Rusty.

### CODE REVIEW

**Comment by patrick-ogrady** [comment]
> You may think that using std::mem::size_of::<u64>() (etc.) is overkill. If so, fair enough. If you want, let me know and I'll change it to a good ol' 8 or 4.

I actually prefer `size_of` and wish we did it elsewhere (magic numbers == bad). Would you mind importing it, so that you can just write `size_of::<u64>()` instead of the full path?

**Comment by patrick-ogrady** [comment]
> In those cases where we care about code aesthetics more than performance, I'd suggest doing, for example

Maybe I have bad taste, but I actually like the `extend_from_slice` look better (I guess its convenient it is also more performant lol).

File: p2p/src/authenticated/actors/tracker/address.rs [hunk 1]
```diff
@@ -18,15 +18,15 @@ pub struct Signature {
...
     payload.extend_from_slice(&peer.timestamp.to_be_bytes());
     payload
 }
 
 pub fn socket_peer_payload(socket: &SocketAddr, timestamp: u64) -> (Vec<u8>, Vec<u8>) {
     let socket = bytes(socket);
-    let mut payload = Vec::new();
+    let mut payload = Vec::with_capacity(socket.len()+std::mem::size_of::<u64>());
     payload.extend_from_slice(&socket);
     payload.extend_from_slice(&timestamp.to_be_bytes());
```
Comment (patrick-ogrady) [inline p2p/src/authenticated/actors/tracker/address.rs:34]: I wonder if we should switch this to `put_u64` (like we do in `mocks/application` above)?

I started using that pattern after writing this crate.

Comment (danlaine) [reply p2p/src/authenticated/actors/tracker/address.rs:34]: Yeah I made a separate branch with those changes (`danlaine/danlaine-put-u64`). I'll put up a PR once this one gets merged.

**Comment by danlaine** [comment]
> I actually prefer `size_of` and wish we did it elsewhere (magic numbers == bad). Would you mind importing it, so that you can just write `size_of::<u64>()` instead of the full path?

Done -- now we import `std::mem::size_of` and just invoke `size_of`

File: p2p/src/authenticated/actors/tracker/address.rs (hunk 1 above)
Comment (patrick-ogrady) [reply p2p/src/authenticated/actors/tracker/address.rs:34]: Appreciate it 🙏

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**